app/scripts/modules/core/src/authentication/userMenu/userMenu.directive.ts

```diff
@@ -0,0 +1,27 @@
+import {AUTHENTICATION_INITIALIZER_SERVICE} from '../authentication.initializer.service';
```

Comment:
Could this be an angular component instead?  Or even a React component?

Reply:
Not without throwing out all the bootstrap CSS (I looked into this a long time ago). Definitely outside the scope of this PR.

Reply:
If it were a React component, though, it would work with the Bootstrap CSS, right?. The only reason we've been stuck with it as a directive and not a component is because of the `replace: true` bit.

Agree it would be nice to cut it to React, but might make sense to do it in a separate PR, as this one's already ~1500 LOC

Reply:
Yea, but 1200 of those are in one file, the styleguide.html....just sayin'

Reply:
heh, fair point, hadn't made it that far in the PR